Cabal/Cabal.cabal

```diff
@@ -263,7 +263,9 @@ library
       Distribution.Compat.Binary.Generic
 
   default-language: Haskell98
-  default-extensions: CPP
+  -- starting with GHC 7.0, rely on {-# LANGUAGE CPP #-} instead
+  if !impl(ghc >= 7.0)
```

Comment:
[Looks like you're right](https://github.com/haskell/cabal/blob/master/Cabal/Distribution/PackageDescription/Configuration.hs#L112), they're different. `impl(ghc < 7)` is basically a shorthand for `impl(ghc) && impl(ghc < 7)`.

Hugs is no longer supported by Cabal, though, and I don't think we expect anything except GHC/GHCJS to be able to compile Cabal.